### DISCUSSION TITLE
I want to extend Hookform . I want to make nestform

### DISCUSSION THREAD

**Discussion by alexk345** [react-hook-form/react-hook-form#11991] [General]
Anyone done nested form? Basically the form will be virtual form. Fieldset will be the form. When getValues called it will ignore child fieldset
form submit only happen via type="submit" onClick  on that button. 

Hookform will be a fieldset component. But i need form outer element with Onsubmit=false.

This way we can as many fieldset nested or not. endless nested form. 

======================
 chunk-GZ55BCQ2.js?v=2e412352:521 Warning: validateDOMNesting(...): <form> cannot appear as a descendant of <form>.
    at form
    at fieldset
    at GenericForm 

========================================    

None of above error message.
  
  I have to ignore this warning and continue my nested form code
  
  But i gave idea on how to do nested form. if someone already done , let me know
  
  react native does that way. Last year i tested nested form with react native and no warning. so when i did with hookform i noticed it was crashing.

**Comment by alexk345** [comment]
Basically nested childform design , Hookchildform work on fieldset level

**Comment by alexk345** [comment]
Can be implmented easily by making prefix on two functions. getValues and SetValue , user can have any prefix. One parent form  and we can have hookchildform which is basically a fieldsetelement. mainly to render only elements inside the fieldset.

**Comment by tatsmaki** [comment]
Based on your requirements there are might be different ways to develop a feature.
As an example – you can implement a single form which wraps multiple form steps using FormContext.
But none of correct solutions will actually nest form tags.

**Comment by alexk345** [comment]
hi

i just ignore all form warning now.  Its annoying but i no one look at
console anyways

[image: image.png]
